#### Q1. What are the three measures of central tendency?

The three measures of central tendency are:
    1. Mean
    2. Median
    3. Mode

.

#### Q2. What is the difference between the mean, median, and mode? How are they used to measure the
central tendency of a dataset?

Mean: The mean is the average of all the values in a dataset. It is calculated by summing up all the values and dividing by the total number of values.

Median: The median is the middle value of a dataset when the values are arranged in ascending or descending order. If there is an even number of values, the median is the average of the two middle values. The median is not affected by extreme values (outliers) in the dataset.

Mode: The mode is the value that occurs most frequently in a dataset. Unlike the mean and median, the mode can be applied to categorical data as well as numerical data. 

.

#### Q3. Measure the three measures of central tendency for the given height data:

In [2]:
x= [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [4]:
import numpy as np
np.mean(x)

177.01875

In [5]:
np.median(x)

177.0

In [8]:
from scipy import stats
stats.mode(x)

ModeResult(mode=177.0, count=3)

.

#### Q4. Find the standard deviation for the given data:

In [9]:
x= [178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [10]:
np.std(x)

1.7885814036548633

.

#### Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe the spread of a dataset? Provide an example.

Range: The range is the simplest measure of dispersion and is calculated as the difference between the maximum and minimum values in the dataset. It gives an indication of the spread of the data across its entire range.

Variance: Variance measures the average squared deviation of each data point from the mean of the dataset. It provides a more precise measure of dispersion compared to the range because it considers the differences of all data points from the mean.

Standard Deviation: The standard deviation is the square root of the variance. It measures the average deviation of data points from the mean and is expressed in the same units as the original data. Standard deviation is widely used because it is more interpretable than variance. A smaller standard deviation indicates that the data points are closer to the mean, while a larger standard deviation indicates greater variability.

In [15]:
# Example
City= [20, 22, 21, 19, 23, 20, 21]

In [16]:
City

[20, 22, 21, 19, 23, 20, 21]

In [18]:
np.max(City)

23

In [19]:
np.min(City)

19

In [20]:
# variance
np.var(City)

1.551020408163265

In [21]:
# Std
np.std(City)

1.2453996981544782

.

#### Q6. What is a Venn diagram?


A Venn diagram is a graphical representation used to illustrate relationships between different sets or groups of items. It consists of circles, typically overlapping, where each circle represents a set and the overlapping areas represent intersections between the sets. The primary purpose of a Venn diagram is to visually demonstrate the similarities, differences, and intersections between various sets or groups of data. They are commonly used in mathematics, logic, statistics, and various fields of study to analyze and compare relationships between different categories or groups.

.

#### Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:
(i) A intersection B

(ii) A ⋃ B

(i). (2,6)

(ii). (0,2,3,4,5,6,7,8,10)

.

#### Q8. What do you understand about skewness in data?


Skewness in data refers to the asymmetry or lack of symmetry in the distribution of values within a dataset. It is a measure of the extent to which the data deviates from a symmetrical distribution around the mean.

.

#### Q9. If a data is right skewed then what will be the position of median with respect to mean?

Right-skewed is define as the longer or fatter tail on the right side of the distribution, the position of the median with respect to the mean will typically be to the left of the mean.

In a right-skewed distribution:
- The tail on the right side contains relatively very large values, which pull the mean towards the right.
- The majority of the data points are concentrated on the left side of the distribution, which affects the median.
- Since the median represents the middle value of the dataset when arranged in ascending or descending order, it is less influenced by extreme values in the right tail compared to the mean.
- Therefore, in a right-skewed distribution, the median will typically be to the left of the mean.

.

#### Q10. Explain the difference between covariance and correlation. How are these measures used in statistical analysis?


Covariance and correlation are both measures used to describe the relationship between two variables in a dataset:

- Covariance measures the degree to which two variables change together. It indicates the direction of the linear relationship between two variables. However, the magnitude of covariance depends on the scale of the variables, making it challenging to interpret.

- Correlation, on the other hand, standardizes covariance by dividing it by the product of the standard deviations of the variables. This results in a correlation coefficient that ranges from -1 to 1. A correlation of 1 indicates a perfect positive linear relationship, -1 indicates a perfect negative linear relationship, and 0 indicates no linear relationship


.

#### Q11. What is the formula for calculating the sample mean? Provide an example calculation for a   dataset.


The formula for calculating the sample mean (average) of a dataset is:

Sample Mean = sum of all the elements / number of elements

In [35]:
# Example
data=[75,82,90,68,88]

In [36]:
# sum of all the elements
sum = 75+82+90+68+88
sum

403

In [37]:
# total element in the data 
total = 5

In [38]:
mean = sum/total

In [39]:
# so the mean of the data is
mean

80.6

.

#### Q12. For a normal distribution data what is the relationship between its measure of central tendency?


For a normal distribution, the relationship between its measures of central tendency (mean, median, and mode) is as follows:

The mean of a normal distribution is equal to its median, and it is also equal to its mode. In other words, in a perfectly symmetrical normal distribution, the mean, median, and mode are all located at the center of the distribution.

Mean = Median = Mode

.

#### Q13. How is covariance different from correlation?

Covariance measures how two variables change together but doesn't indicate the strength or direction of the relationship.

Correlation is a standardized measure of covariance that ranges from -1 to 1, indicating the strength and direction of the linear relationship between variables

.

#### Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.

Outliers can heavily influence the mean, pulling it towards them and distorting it.
The median is less affected by outliers because it represents the middle value.
Outliers can greatly impact the range, variance, and standard deviation, inflating these measures and overestimating the spread of the data.

In [41]:
# Example

In [42]:
data = [20,25,30,35,40,45,50,55,60,1000]

##### mean , median and mode Without the outliers  (1000)

In [47]:
data = [20,25,30,35,40,45,50,55,60]
print("Mean without outliers present = ",np.mean(data))
print("Median without outliers present = ",np.median(data))
print("Mode without outliers present = ",stats.mode(data))

Mean without outliers present =  40.0
Median without outliers present =  40.0
Mode without outliers present =  ModeResult(mode=20, count=1)


##### mean, median and mode with the outliers (1000)

In [48]:
data = [20,25,30,35,40,45,50,55,60,1000]
print("Mean when outliers is present = ",np.mean(data))
print("Median when outliers is present = ",np.median(data))
print("Mode when outliers is present = ",stats.mode(data))

Mean when outliers is present =  136.0
Median when outliers is present =  42.5
Mode when outliers is present =  ModeResult(mode=20, count=1)


- without the outlier the mean is 40 and with the outliers the mean becomes 136 which is significantly higher.
- The median is 40 without the outlier, and it remains 42 even with the outlier because it is not affected by extreme values.
- The mode remains 20 without the outlier, and it doesn't change even with the outlier

Range: Without the outlier, the range is 60 − 20 = 40
but with the outlier, it becomes  1000−20= 980.

Variance and Standard Deviation: Both the variance and standard deviation increase significantly with the outlier, indicating a larger spread of the data.